src/rust/ensogl/src/display/shape/text/text_field/cursor.rs

```diff
@@ -599,10 +593,12 @@ mod test {
         expected_positions.insert(DocBegin,  vec![(0,0)]);
         expected_positions.insert(DocEnd,    vec![(2,9)]);
 
-        let mut fonts      = FontRegistry::new();
-        let properties     = TextFieldProperties::default(&mut fonts);
-        let mut content    = TextFieldContent::new(text,&properties);
-        let mut navigation = CursorNavigation::new(&mut content);
+        let mut fonts       = FontRegistry::new();
+        let properties      = TextFieldProperties::default(&mut fonts);
+        let content         = &mut TextFieldContent::new(text,&properties);
+        let text_field_size = properties.size;
+        let selecting       = false;
+        let mut navigation  = CursorNavigation{selecting,content,text_field_size};
 
         for step in &[/*Left,Right,Up,*/Down,/*LineBegin,LineEnd,DocBegin,DocEnd*/] {
```

Comment:
Oh, now I see that I accidentally push commented cases! Please uncomment them and add cases for PageUp and PageDown. It covers more corner cases (like cursor in the first line, in the last line, at end of line, at end of document etc.)